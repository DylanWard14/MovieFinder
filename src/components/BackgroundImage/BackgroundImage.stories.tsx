import React from 'react';
import {withKnobs, text} from '@storybook/addon-knobs';

import BackgroundImage from './BackgroundImage';

export default {
    title: 'BackgroundImage',
    decorators: [withKnobs]
};

export const DefaultExample = () => (<BackgroundImage imageURLExtension="/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg" />);
export const PropsExample = () => <BackgroundImage imageURLExtension={text("URL Extenstion", "/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg")}/>;