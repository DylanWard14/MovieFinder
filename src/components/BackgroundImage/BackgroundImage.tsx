import React from 'react';
import styled from 'styled-components/macro';

interface BackgroundImageProps {
    imageURLExtension: string;
    children?: React.ReactNode | React.ReactNode[];
}

const Image = styled.div<{ url: string }>`
    height: 100vh;
    background-image: linear-gradient(
            to bottom,
            rgba(0, 0, 0, 1),
            rgba(0, 0, 0, 0.75),
            rgba(0, 0, 0, 0.1),
            rgba(0, 0, 0, 0.75),
            rgba(0, 0, 0, 1)
        ),
        url(${props => props.url});
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    box-sizing: border-box;

    padding: 1rem;
`;

const BackgroundImage: React.FunctionComponent<BackgroundImageProps> = ({ imageURLExtension, children }) => {
    const FullImageUrl = `https://image.tmdb.org/t/p/original${imageURLExtension}`;

    return <Image url={FullImageUrl}>{children}</Image>;
};

export default BackgroundImage;
