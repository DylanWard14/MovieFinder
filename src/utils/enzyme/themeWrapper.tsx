import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
