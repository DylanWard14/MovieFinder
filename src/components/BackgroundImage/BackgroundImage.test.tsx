import React from 'react';

import { renderWrapper } from '../../utils/enzyme/themeWrapper';
import BackgroundImage from './BackgroundImage';

describe('BackgroundImage', () => {
    it('renders with a background image correctly', () => {
        const wrapper = renderWrapper(<BackgroundImage imageURLExtension="/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg" />);
        expect(wrapper).toMatchSnapshot();
    });
});
