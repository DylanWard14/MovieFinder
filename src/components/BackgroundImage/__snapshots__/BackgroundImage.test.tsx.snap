// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage renders with a background image correctly 1`] = `
<div
  class="BackgroundImage__Image-sc-1a331ab-0 fDJSrr"
/>
`;
